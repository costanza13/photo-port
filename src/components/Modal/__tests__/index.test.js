import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const currentPhoto = {
  name: 'Grocery aisle',
  category: 'commercial',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
  index: 1
};

const mockToggleModal = jest.fn();

afterEach(cleanup);

describe('Modal is rendering', () => {
  it('renders', () => {
    render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModal} />);
  });
});

describe('Modal displays image', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModal} />)
    expect(asFragment()).toMatchSnapshot()
  });
});

describe('Modal displays metadata', () => {
  it('renders', () => {
    const { getByTestId } = render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModal} />)
    expect(getByTestId('modalPhotoName')).toHaveTextContent('Grocery aisle')
    expect(getByTestId('modalPhotoDescription')).toHaveTextContent('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie')
    expect(getByTestId('modalCloseButton')).toHaveTextContent('Close this modal')
  });
});

describe('Click Event', () => {
  it('calls onClose handler', () => {
    const { getByText } = render(<Modal currentPhoto={currentPhoto} onClose={mockToggleModal} />)
    fireEvent.click(getByText('Close this modal'));
    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  });
}) 