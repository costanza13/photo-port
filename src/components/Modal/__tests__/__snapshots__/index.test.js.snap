// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal displays image matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
        data-testid="modalPhotoName"
      >
        Grocery aisle
      </h3>
      <img
        alt="commercial"
      />
      <p
        data-testid="modalPhotoDescription"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        data-testid="modalCloseButton"
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
